tin/apps/assignments/templatetags/index.py

```diff
@@ -0,0 +1,13 @@
+from typing import Mapping, TypeVar
+
+from django import template
+
+register = template.Library()
+
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+@register.filter
+def index(indexable: Mapping[T, U], index: T) -> U:
+    return indexable[index]
```

Comment:
Cool template tag implementation, but I don't think it's the ideal solution here (readability takes a big hit). I would suggest considering the following idea:

`AssignmentForm.sections` stays how it is, returning all sections
`AssignmentForm.main_section` (or similar) returns the required section

Alternatively, my previous suggestion for collapsible and uncollapsible sections would fix this problem by just having everything in a section.

Reply:
> Alternatively, my previous suggestion for collapsible and uncollapsible sections would fix this problem by just having everything in a section.

I'm not sure that I understand how this eliminates the need for a filter. The solution you proposed would still require indexing a dictionary (to find, say, if it's collapsible or not), which (to my knowledge) you cannot do in django templates without a custom filter.
Could you possibly elaborate on that?

Reply:
If I recall correctly, dot notation works with dictionary lookups. The reason it doesn't currently work is due to variable dictionary keys (currently, they're the names of the sections). If the data you want is guaranteed to be in the dictionary's `fields` key, however, then `field_dict.fields` should just work...

Reply:
Oh interesting, never knew that. With some preprocessing in the `get_section` methods I pushed a commit fixing it.